src/lib/viz/style/Style.ts

```diff
@@ -1,18 +1,27 @@
 import { GeoJsonLayerProps } from '@deck.gl/layers/geojson-layer/geojson-layer';
+import { LegendProperties } from '@/viz/legend';
 import { CartoStylingError, stylingErrorTypes } from '../errors/styling-error';
 import { StyledLayer } from './layer-style';
 
 export type StyleProperties =
   | GeoJsonLayerProps<any>
   | ((layerStyle: StyledLayer) => GeoJsonLayerProps<any>);
 
+type LegendPropertiesFunction = (layerStyle: StyledLayer, options: any) => LegendProperties[];
```

Comment:
IMO after the rename one part could stay in plural and the type in singular, like:
```
type LegendPropertiesFunction = (layerStyle: StyledLayer, options: any) => LegendProperty[];
```

What do you think?